src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclint/Checker.java

```diff
@@ -884,28 +856,26 @@ public Void visitLiteral(LiteralTree tree, Void ignore) {
     }
 
     @Override @DefinedBy(Api.COMPILER_TREE)
-    @SuppressWarnings("fallthrough")
     public Void visitParam(ParamTree tree, Void ignore) {
         boolean typaram = tree.isTypeParameter();
         IdentifierTree nameTree = tree.getName();
         Element paramElement = nameTree != null ? env.trees.getElement(new DocTreePath(getCurrentPath(), nameTree)) : null;
 
         if (paramElement == null) {
             switch (env.currElement.getKind()) {
-                case CLASS: case INTERFACE: {
-                    if (!typaram) {
+                case CLASS, INTERFACE -> {
```

Comment:
A record can be generic too.

Reply:
Yes, but RECORD is handled in a different case, along with METHOD and CONSTRUCTOR, because it can have "plain" `@param` as well as type-parameter `@param`.  In contrast, a class or interface can only have type-parameter `@param` and never plain `@param`.

Reply:
> Yes, but RECORD is handled in a different case, along with METHOD and CONSTRUCTOR, because it can have "plain" `@param` as well as type-parameter `@param`. In contrast, a class or interface can only have type-parameter `@param` and never plain `@param`.

You are correct; I missed that unconditionality in that second arm for`METHOD, CONSTRUCTOR, RECORD`.